### PR TITLE
fix(check): filter staged files per-step in pre-commit orchestrator [OS-450]

### PR DESCRIPTION
**Pull request by galligan**
## Problem

The pre-commit check orchestrator fails with "Expected at least one target file" when only non-JS/TS files are staged (e.g., `.sh`, `.yml`, `.md` files). This blocks shell-only or config-only commits from going through without bypassing hooks.

**Discovered in:** #640 — committing `agent-setup.sh` and `agent-maintenance.sh` required `LEFTHOOK=0` to bypass.

## Root Cause

The `ultracite-fix` step in `buildCheckOrchestratorPlan` passed all `stagedFiles` to ultracite without filtering by supported extension. When only non-JS/TS files were staged, ultracite found zero valid targets and exited non-zero, crashing the orchestrator.

## Fix

Filter staged files for the `ultracite-fix` step to only include supported extensions (`.js`, `.jsx`, `.ts`, `.tsx`, `.json`, `.jsonc`, `.css`), mirroring the existing `tsFiles` pattern used for typecheck. The step is skipped entirely when no supported files are present among staged files.

### Changes

- **`apps/outfitter/src/commands/check-orchestrator.ts`**: Added `ultraciteFiles` filter; conditionally include `ultracite-fix` step only when there are matching files (or no staged files for full-tree fallback)
- **`apps/outfitter/src/__tests__/check-orchestrator.test.ts`**: Updated existing test, added tests for shell-only staged files and mixed file commits

### Edge Cases Covered

| Scenario | Behavior |
|---|---|
| Mixed `.sh` + `.ts` | Only `.ts` passed to ultracite |
| All non-JS staged files | `ultracite-fix` step skipped |
| No staged files | Falls back to `.` (full-tree check) |
| `.json`/`.jsonc` only | Included in ultracite filter |

Closes OS-450

### CODE REVIEW

File: apps/outfitter/src/commands/check-orchestrator.ts [hunk 1]
```diff
@@ -329,21 +329,31 @@ export function buildCheckOrchestratorPlan(
 
   const hasStagedFiles = stagedFiles.length > 0;
   const tsFiles = stagedFiles.filter((f) => /\.(ts|tsx)$/.test(f));
+  const ultraciteFiles = stagedFiles.filter((f) =>
+    /\.(js|jsx|ts|tsx|json|jsonc|css)$/.test(f)
+  );
```
[the remark this answers is not in the document]
Comment (galligan) [reply apps/outfitter/src/commands/check-orchestrator.ts]: Fixed in 94d487e — added `.mjs`, `.cjs`, `.mts`, `.cts` to the filter regex. Test updated to include an `.mjs` file in the mixed-commit scenario.

**Comment by galligan** [comment]
* **#641** <a href="https://app.graphite.com/github/pr/outfitter-dev/outfitter/641?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈 <a href="https://app.graphite.com/github/pr/outfitter-dev/outfitter/641?utm_source=stack-comment-view-in-graphite" target="_blank">(View in Graphite)</a>
* `main`


<h2></h2>

This stack of pull requests is managed by <a href="https://graphite.dev?utm-source=stack-comment"><b>Graphite</b></a>. Learn more about <a href="https://stacking.dev/?utm_source=stack-comment">stacking</a>.
